File: split/confusion.pas
procedure confusion(s: strnumber);
begin
  normalizeselector;
  if history < 2 then
  begin
    begin
      if interaction = 3 then;
      printnl(262);
      print(291);
    end;
    print(s);
    printchar(41);
    begin
      helpptr := 1;
      helpline[0] := 292;
    end;
  end
  else
  begin
    begin
      if interaction = 3 then;
      printnl(262);
      print(293);
    end;
    begin
      helpptr := 2;
      helpline[1] := 294;
      helpline[0] := 295;
    end;
  end;
  begin
    if interaction = 3 then
      interaction := 2;
    if logopened then
      error;
    {if interaction>0 then debughelp;}
history := 3;
    jumpout;
  end;
end;

File: cpp/confusion.cpp
#include "confusion.h"
#include "normalizeselector.h"
#include "printnl.h"
#include "print.h"
#include "printchar.h"
#include "error.h"
#include "jumpout.h"
#include "texte.h"

void confusion(const std::string &s)
{
	normalizeselector();
	if (history < error_message_issued)
	{
		printnl("! ");
		print("This can't happen (");
		print(s);
		printchar(')');
		helpptr = 1;
		helpline[0] = txt("I'm broken. Please show this to someone who can fix can fix");
	}
	else
	{
		printnl("! ");
		print("I can't go on meeting you like this");
		helpptr = 2;
		helpline[1] = txt("One of your faux pas seems to have wounded me deeply...");
		helpline[0] = txt("in fact, I'm barely conscious. Please fix it and try again.");
	}
	if (interaction == error_stop_mode)
		interaction = scroll_mode;
	if (logopened)
		error();
	history = fatal_error_stop;
	jumpout();
}
